src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -191,6 +193,22 @@ public override void Close()
 		}
 #endif
 
+		public override T GetFieldValue<T>(int ordinal)
+		{
+			if (typeof(T) == typeof(DateTimeOffset))
+				return (T) Convert.ChangeType(GetDateTimeOffset(ordinal), typeof(T));
+
+			return base.GetFieldValue<T>(ordinal);
+		}
+
+		public override async Task<T> GetFieldValueAsync<T>(int ordinal, CancellationToken cancellationToken)
+		{
+			if (typeof(T) == typeof(DateTimeOffset))
+				return (T) Convert.ChangeType(GetDateTimeOffset(ordinal), typeof(T));
```

Comment:
Consider using `Task.FromResult` here to avoid an unnecessary `await` below. (Stephen Cleary has some thoughts on that [here](http://blog.stephencleary.com/2016/12/eliding-async-await.html) but I think it's worth it in this case since `GetFieldValueAsync` ought to be low-overhead.)